### DISCUSSION TITLE
Hocuspocus server + Lexical compatibility

### DISCUSSION THREAD

**Discussion by vineetdigit** [ueberdosis/hocuspocus#597] [General]
Hello,

We're looking at hocuspocus as a collab backend for lexical. Although it appears on the face of it that everything should work out of the box, our initial experiments resulted in backend throwing errors. All we did was to replace the yjs-websocket server with @hocuspocus/server, but did not change anything on the lexical.

> TypeError [ERR_ENCODING_INVALID_ENCODED_DATA]: The encoded data was not valid for encoding utf-8

Does hocuspocus work with y-websocket provider that's integrated in lexical or does it require using hocuspocus provider?

thx,

Vineet

**Comment by janthurau** [comment]
hey @vineetdigit , we're not compatible with y-websocket provider since v2.0, because we've added multiplexing support. Can you switch to our HocuspocusProvider?

**Reply by vineetdigit** [reply]
Ok thanks for your reply. I like the fact that hocuspocus introduced multiplexing even if it meant breaking the compatibility. Clearly that is how y-websocket provider should've been done to start with. But what this means for us in the short term is to rework lexical's bindings to use the hocuspocus provider. It seems to have similar api as the y-websocket provider, but it's not a drop-in replacement. thanks again!

**Reply by janthurau** [reply]
@vineetdigit you don't need to remove the lexical binding (lexical <> ydoc), just the part that transfers ydoc <> hocuspocus needs to be replaced. We actually have another user who recently had an issue with lexical, maybe this repo here helps you: https://github.com/janthurau/egonbolton-collab
